float32[4] rads_filtered
float32[4] rads_raw